### PR TITLE
Fix issue where language wasn't set according to the manifest

### LINKED ISSUE #291: UIs not respecting the locale specified in the manifest
Please see https://spid.zendesk.com/agent/tickets/17453 for details

Default locale works but ends up being English when Norwegian is specified

### PR DESCRIPTION
**Pull request by Astach**
Fixes #291

### CODE REVIEW

File: ui/src/main/java/com/schibsted/account/ui/OptionalConfiguration.kt [hunk 1]
```diff
@@ -76,7 +76,10 @@ data class OptionalConfiguration(
             val keyIsCancellable = appContext.getString(R.string.schacc_conf_cancellable)
             val keyClientLogo = appContext.getString(R.string.schacc_conf_client_logo)
 
-            val locale: Locale? = appInfo.metaData.getString(keyLocale)?.let { Locale(it) }
+            val locale: Locale? = appInfo.metaData.getString(keyLocale)?.let {
+                val (lang, country) = it.split("_")
```
Comment (Nexcius) [inline ui/src/main/java/com/schibsted/account/ui/OptionalConfiguration.kt]: Please add a sanity check to this to output a proper error message

File: ui/src/main/java/com/schibsted/account/ui/login/BaseLoginActivity.kt [hunk 2]
```diff
@@ -156,6 +157,10 @@ abstract class BaseLoginActivity : AppCompatActivity(), KeyboardManager, Navigat
         }
     }
 
+    override fun attachBaseContext(base: Context) {
+        super.attachBaseContext(UiUtil.updateContextLocale(base, OptionalConfiguration.fromManifest(base).locale))
```
Comment (Nexcius) [inline ui/src/main/java/com/schibsted/account/ui/login/BaseLoginActivity.kt]: :+1:

File: ui/src/main/java/com/schibsted/account/ui/UiUtil.kt [hunk 3]
```diff
@@ -288,6 +288,22 @@ object UiUtil {
...
+    /**
+     * @param localeTag a tag representing a locale
+     * @return a Locale if the tag format is correct null otherwise
+     * @see https://developer.android.com/reference/java/util/Locale
+     */
+    fun getLocaleFromLocaleTag(localeTag: String): Locale? {
+        val localeInfo = localeTag.split("_")
+        if (localeInfo.size == 2) {
+            val (lang, country) = localeInfo
+            if ((lang.length in 2..3) && country.length == 2) {
```
[the remark this answers is not in the document]
Comment (Astach) [reply ui/src/main/java/com/schibsted/account/ui/UiUtil.kt]: I do not see why doing the proper check is overkill.
`n_No` is not correct, the language would be set to `Locale.getDefault()` 
`nb_n` is not correct, I do not know how n would be interpreted by Spid.
Also checking that the format is correct could avoid us to do some support

[the remark this answers is not in the document]
Comment (Nexcius) [reply ui/src/main/java/com/schibsted/account/ui/UiUtil.kt]: When you construct a locale, either `Locale(NB, NO)` and `Locale(nb, no)` will result in it's toString()  being `nb_NO`

[the remark this answers is not in the document]
Comment (Astach) [reply ui/src/main/java/com/schibsted/account/ui/UiUtil.kt]: I don't check if it's capitalized, I don't follow you

**Pull request merged**